ChangeLog.d/gcc-14.3.txt

```diff
@@ -0,0 +1,8 @@
+Changes
+   * Move definition of MBEDTLS_HAS_BUILTIN macro earlier in
+     core/tf_psa_crypto_common.h.
+
```

Comment:
That's a minor thing that I don't think it's worth being mentioned in changelog. But that's my opinion though, please wait for other reviewer's feedback before removing it

Reply:
Ah sorry I did not specify, my bad! The ChangeLog-worthy thing here would be something like "Fix warning about array bounds in mbedtls_xor() when building with GCC 14.3". Would you be able to put something like that instead?